### PR TITLE
Bug fix in swap: directed_edge_swap and double_edge_swap 

### LINKED ISSUE #6144: directed_edge_swap: ZeroDivisionError when passing a digraph without edges
I was working on improving test coverage for `swap.py` and tested `directed_edge_swap` with a digraph without edges. That resulted in a ZeroDivisionError from `nx.utils.cumulative_distribution`. 

### Steps to Reproduce

```python
from networkx as nx

G = nx.DiGraph()
G.add_nodes_from([0, 2, 4, 3])

G = nx.directed_edge_swap(G, nswap=1, max_tries=500, seed=1)
```
### Current Behavior

ZeroDivisionError is raised because in `nx.utils.cumulative_distribution` there's a division by the sum of the input distribution that is zero. The distribution that `nx.utils.cumulative_distribution` is called with in this example is the degrees of a digraph without edges (all degrees are zero). This is the error log. 

```
Traceback (most recent call last):
  File "test.py", line 12, in <module>
    G = nx.directed_edge_swap(G, nswap=1, max_tries=500, seed=1)
  File "/home/paula/Outreachy/networkX/networkx/networkx/utils/decorators.py", line 766, in func
    return argmap._lazy_compile(__wrapper)(*args, **kwargs)
  File "<class 'networkx.utils.decorators.argmap'> compilation 21", line 5, in argmap_directed_edge_swap_17
  File "/home/paula/Outreachy/networkX/networkx/networkx/algorithms/swap.py", line 81, in directed_edge_swap
    cdf = nx.utils.cumulative_distribution(degrees)  # cdf of degree
  File "/home/paula/Outreachy/networkX/networkx/networkx/utils/random_sequence.py", line 103, in cumulative_distribution
    cdf.append(cdf[i] + distribution[i] / psum)
ZeroDivisionError: division by zero
```
### Expected Behavior

I think the best is to raise an exception in `directed_edge_swap` with a message that exposes this situation.  Also, this should be added to the [doc entry](https://output.circle-artifacts.com/output/job/ab098af6-f76c-4bf9-a732-5c243aca96da/artifacts/0/doc/build/html/reference/algorithms/generated/networkx.algorithms.swap.directed_edge_swap.html#directed-edge-swap). 

### Environment

Python version: 3.8
NetworkX version: 2.8.7

### Further Discoveries

Something else that should be added is that if the digraph has less than 3 edges then is impossible to swap edges. In that case, the current behavior is that the max number of tries is reached. In my opinion, this case should raise an exception with an appropriate message. Also, this could be mentioned in the  [doc entry](https://output.circle-artifacts.com/output/job/ab098af6-f76c-4bf9-a732-5c243aca96da/artifacts/0/doc/build/html/reference/algorithms/generated/networkx.algorithms.swap.directed_edge_swap.html#directed-edge-swap). 

I will work on these changes and will look closely at the other functions in `swap.py`. 
Also, I think it will be good to review the functions that use `nx.utils.cumulative_distribution` to check for this error. I can do that too.

### PR DESCRIPTION
**Pull request by paulitapb**
Closes #6144 
Changed directed_edge_swap, now an exception is raised if the number of edges is less than 3. Also updated the documentation.

Changed `double_edge_swap`, now an exception is raised when the function is called with a graph with fewer than 2 edges. Added some tests and updated the documentation to include a raises section that was missing.

### CODE REVIEW

File: networkx/algorithms/swap.py [hunk 1]
```diff
@@ -161,6 +163,15 @@ def double_edge_swap(G, nswap=1, max_tries=100, seed=None):
     G : graph
        The graph after double edge swaps.
 
+    Raises
+    ------
+    NetworkXError
+        If `G` is directed, or
+        If nswap > max_tries, or
```
Comment (rossbar) [inline networkx/algorithms/swap.py:170]:
```suggestion
        If `nswap` > `max_tries`, or
```

File: networkx/algorithms/swap.py [hunk 2]
```diff
@@ -70,7 +70,9 @@ def directed_edge_swap(G, *, nswap=1, max_tries=100, seed=None):
     if nswap > max_tries:
         raise nx.NetworkXError("Number of swaps > number of tries allowed.")
     if len(G) < 4:
-        raise nx.NetworkXError("Graph has less than four nodes.")
+        raise nx.NetworkXError("DiGraph has less than four nodes.")
+    if len(G.edges) < 3:
+        raise nx.NetworkXError("DiGraph has less than 3 edges")
```
Comment (rossbar) [inline networkx/algorithms/swap.py:75]: Again, very minor grammatical nit: less -> fewer (might as well take the opportunity to clean up old grammar!)

File: networkx/algorithms/tests/test_swap.py [hunk 3]
```diff
@@ -123,3 +123,18 @@ def test_degree_seq_c4():
...
+
+
+def test_no_edges():
+    G = nx.DiGraph()
+    G.add_nodes_from([0, 1, 2])
+    pytest.raises(nx.NetworkXError, nx.directed_edge_swap, G)
+    G = nx.Graph()
+    G.add_nodes_from([0, 1, 2, 3])
+    pytest.raises(nx.NetworkXError, nx.double_edge_swap, G)
+
```
Comment (rossbar) [inline networkx/algorithms/tests/test_swap.py:135]: Just another minor suggestion: using `pytest.raises` as a context manager with the `match=` check is a really nice feature that allows us to better test we're hitting the exact exception branch that we expect.

File: networkx/algorithms/tests/test_swap.py [hunk 4]
```diff
@@ -123,3 +123,18 @@ def test_degree_seq_c4():
...
+
+def test_less_than_3_edges():
+    G = nx.DiGraph()
+    G.add_edges_from([(0, 1), (1, 2)])
+    pytest.raises(nx.NetworkXError, nx.directed_edge_swap, G)
```
Comment (rossbar) [inline networkx/algorithms/tests/test_swap.py:140]:
```suggestion
    G = nx.DiGraph([(0, 1), (1, 2)])
    with pytest.raises(nx.NetworkXError, match=".*fewer than 3 edges"):
        nx.directed_edge_swap(G)
```

Note that the below wording for `match=` is predicated on the assumption that we update 'less' -> 'fewer' in the exception message.

**Review by rossbar — APPROVED** [review]
LGTM - I reviewed the `directed_edge_swap` and `double_edge_swap` functions and I agree with your assessment that they are expected to fail if there are fewer than 3 or 2 edges, respectively. The explicit checking and nicer exception messages are a nice improvement!

All the suggestions are on the very nit-picky end of the spectrum, so no blockers. Thanks @paulitapb

**Comment by MridulS** [comment]
@paulitapb just a ping to review the comments :)

File: networkx/algorithms/tests/test_swap.py (hunk 3 above)
Comment (paulitapb) [reply networkx/algorithms/tests/test_swap.py:135]: great!

**Pull request merged**